frontend/packages/kubevirt-plugin/integration-tests-cypress/view/wizard.ts

```diff
@@ -1,14 +1,70 @@
 import { ProvisionSource } from '../enums/provisionSource';
+import { VirtualMachineData } from '../types/vm';
+import { addDisk, addNIC } from './dialog';
+
+const fillBootSource = (
+  provisionSource: ProvisionSource,
+  pvcName: string,
+  pvcNS: string,
+  pvcSize?: string,
+) => {
+  cy.get('#image-source-type-dropdown').click();
+  cy.get('.pf-c-select__menu')
+    .contains(provisionSource.getDescription())
+    .click({ force: true });
+  switch (provisionSource) {
+    case ProvisionSource.URL: {
+      cy.get('input[id="provision-source-url"]').type(provisionSource.getSource());
+      if (pvcSize) {
+        cy.get('#request-size-input')
+          .clear()
+          .type(pvcSize);
+      }
+      break;
+    }
+    case ProvisionSource.CLONE_PVC: {
+      cy.get('body').then(($body) => {
+        if ($body.find('.pf-c-wizard__nav-list').length) {
+          cy.get('button[id="clone-pvc-ns"]').click();
+          cy.get(`a[id="${pvcNS}-Project-link"]`).click();
+          cy.contains('.pf-c-dropdown__toggle-text', 'Select Persistent Volume Claim').click({
+            force: true,
+          });
+          cy.get(`a[id="${pvcName}-PersistentVolumeClaim-link"]`).click();
+        } else {
+          cy.get('button[id="pvc-ns-dropdown"]').click();
+          cy.get(`a[id="${pvcNS}-Project-link"]`).click();
+          cy.get('button[id="pvc-name-dropdown"]').click();
+          cy.get(`a[id="${pvcName}-PersistentVolumeClaim-link"]`).click();
+        }
+      });
+      break;
+    }
+    case ProvisionSource.REGISTRY: {
+      cy.get('input[id="provision-source-container"]').type(provisionSource.getSource());
+      if (pvcSize) {
+        cy.get('#request-size-input')
+          .clear()
+          .type(pvcSize);
+      }
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+};
 
 export const wizard = {
   vm: {
     open: () => {
       cy.byLegacyTestID('item-create').click();
       cy.byLegacyTestID('vm-wizard').click();
     },
-    processSelectTemplate: (name: string) => {
+    processSelectTemplate: (vmData: VirtualMachineData) => {
```

Comment:
I think we should give clearer names to these `process*` functions to better indicate what they're doing. I think this one should just be `selectTemplate` and the ones for tabs can be `fill<tab name>Fields` like `fillGeneralTabFields`.

Reply:
We use `process` since protractor, because it's not only about filling fields, it also has some checkbox and dropdown list.

Reply:
Form fields aren't just text fields. They're any type of input field, so checkboxes and dropdowns count, too.

Reply:
Maybe just ` processSelectTemplate` change to `selectTemplate` is better, for others, like `processBootSource`, 'processReview` etc, in my opinion, `process` is better than `fill`. :)

Reply:
I disagree. I had no idea what `processGeneral` meant until I looked at the function. `process` is a very generic term and doesn't adequately communicate what is happening. The functions are literally manipulating the browser to fill in the form fields, so it makes much more sense to me to explicitly state what they are doing in the function name rather than relying on a vague term.

Reply:
In protractor tests, it's naming `processGeneralStep`/`processNetworkStep`, does it look good to you?
`process` is better than `fill` because it's a step, you can process a step but not fill a step.

If we look at the `select template` page, it says
```
Select a template
The virtual machine can be customized in the next step.
```

Reply:
The names suggested above were `processX` not `processXStep` which is even more ambiguous, but `processXStep` still doesn't tell me exactly what's going on inside the function. My suggestion wasn't `fillXStep`. It was `fillXTabFields`. It clearly explains that it's _filling_ the _fields_ for a given named _tab_. There's no ambiguity about it. `processXStep` could refer to something that's not even UI-related.

Also, I understand that `processXStep` was used in the protractor tests, but that alone doesn't make it a good naming convention and it's something I'd like to correct while we're touching this code.

Reply:
I don't like `fillXTabFields` because it's not a tab and not only fields, on network/storage step it also has addNIC/addDisk. Do you accept if it's `fillBootsource`/`fillGeneral`?

Reply:
NAME is a hard thing. ")

Reply:
Indeed it is. =) It's so hard that there's even a pretty famous quote (in software development) about it:

> There are only two hard things in Computer Science: cache invalidation and naming things.
> -- Phil Karlton

It looks like you're also filling the boot source in the general tab, so for now if you'll change it to `fillBootSourceFields` and for anything else that starts with `process`, add `Tab` or `Modal` or whatever it is you're processing (since neither `processGeneral` or `fillGeneral` make any sense) I'll lgtm it so we can get it merged today. 

I'll think more on the naming since I'll likely be opening a dedicated refactoring PR at a later point to address some of my nitpicky issues.

Thank you for engaging in this back and forth with me. =)